### PR TITLE
LDAP: nitpicks

### LINKED ISSUE #18295: Small corrections for the LDAP modules
During the release we were focusing on the logic. There is some inconsistencies left with the naming and some comments for the methods could be updated

### PR DESCRIPTION
**Pull request by markelog**
* Add more tests

* Correct and clarify comment for Login() method

* Rename methods (hail consistency!)

* Uppercases first letter of the logs everywhere

* Moves method definitions around to more appropriate places

Fixes #18295

### CODE REVIEW

File: pkg/services/ldap/ldap.go [hunk 1]
```diff
@@ -145,29 +146,36 @@ func (server *Server) Close() {
 
 // Login the user.
 // There is several cases -
```
Comment (gotjosh) [inline pkg/services/ldap/ldap.go]:
```suggestion
// There are several cases -
```

File: pkg/services/ldap/ldap.go [hunk 2]
```diff
@@ -145,29 +146,36 @@ func (server *Server) Close() {
...
-// --
-// After either first or second step is done we find the user DN
-// by its username, after that, we then combine it with user password and
-// then try to authentificate that user
+// 1. "admin" user
+// Bind the "admin" user (defined in grafana config file) which has the search privileges
+// in LDAP server, then we search the targeted user through that bind, then the second
+// perform the bind via passed login/password.
+// 2. Single bind
+// If all the users are meant to use with grafana has ability to search in LDAP server
```
Comment (gotjosh) [inline pkg/services/ldap/ldap.go]:
```suggestion
// If all the users meant to be used with Grafana have the ability to search in LDAP server
```

File: pkg/services/ldap/ldap.go [hunk 3]
```diff
@@ -145,29 +146,36 @@ func (server *Server) Close() {
...
-// That user should have search privileges.
-// 2. For some configurations it is allowed to search the
-// user without any authenfication, in such case we
-// perform "unauthenticated bind".
-// --
-// After either first or second step is done we find the user DN
-// by its username, after that, we then combine it with user password and
-// then try to authentificate that user
+// 1. "admin" user
+// Bind the "admin" user (defined in grafana config file) which has the search privileges
```
Comment (gotjosh) [inline pkg/services/ldap/ldap.go]:
```suggestion
// Bind the "admin" user (defined in Grafana config file) which has the search privileges
```

File: pkg/services/ldap/ldap.go [hunk 4]
```diff
@@ -145,29 +146,36 @@ func (server *Server) Close() {
...
+// in LDAP server, then we search the targeted user through that bind, then the second
+// perform the bind via passed login/password.
+// 2. Single bind
+// If all the users are meant to use with grafana has ability to search in LDAP server
+// then we bind with LDAP server with targeted login/password
+// and then search for the said user in order to retrive all the information about them
+// 3. Unauthenticated bind
+// For some LDAP configurations it is allowed to search the
+// user without login/password binding with LDAP server, in such case
+// we will perform "unauthenticated bind" and then search for the
```
Comment (gotjosh) [inline pkg/services/ldap/ldap.go]:
```suggestion
// we will perform "unauthenticated bind", then search for the
```

**Review by gotjosh — APPROVED** [review]
A big 👍 from me! My comments are just minor edits on the wording - feel free to include whichever you see fit as I don't need to see it again. :shipit: ❤️

**Review by torkelo — APPROVED** [review]
great!

**Pull request merged**